### ISSUE TITLE
count number 1D-2D connections in a calculation cell

### ISSUE THREAD

**Issue by caspervdw** [nens/threedigrid-builder#264]
@Jonasvsl commented on [Mon Oct 24 2022](https://github.com/nens/3Di-modeller-interface-installer/issues/24)

---

@Jonasvsl commented on [Mon Oct 24 2022](https://github.com/nens/3Di-modeller-interface-installer/issues/24#issuecomment-1288702819)

not sure if this can be in the checker but we need to have a warning in case there are more than X 1D-2D connections in the same computational cell.

Let's discuss with Nici what would be a treshold value

**Comment by Jonasvsl** [comment]
As discussed with Casper, it is very difficult to make this part of the checker, but we can make this part of the grid builder.

Here's a thought: Why don't we include creating the computational grid in the workflow for the users before uploading the schematisation?

Perhaps even create a workflow with checks and computational grid schematisation and one without for those that feel lucky?
@leendertvanwolfswinkel What are your thoughts on this?

**Comment by leendertvanwolfswinkel** [comment]
I think it is an excellent idea. Very similar to how the schematisation checker is part of the upload wizard. We have just removed the raster checker; in its place we could generate the computational grid and show any errors & warnings from that process in a table. 

Would be helpful if `threedigrid_builder` outputs errors and warnings in a similar way as `threedi_modelchecker`

**Comment by leendertvanwolfswinkel** [comment]
https://github.com/nens/threedi-api-qgis-client/issues/429

**Comment by elisallenens** [comment]
@leendertvanwolfswinkel this ticket appears to consist of two parts:
- handle logs (warnings, errors) from threedi-gridbuilder in the qgis plugin and present them to the user, as is done with the modelchecker
- generate a warning for a number of 1D-2D connections in a calculation cell greater than a certain value

Casper suggests that maybe you could open a separate ticket for the first part?

**Comment by elisallenens** [comment]
@nvolp could you supply a threshold for a maximum value for the 1D-2D connections in a calculation cell?

**Comment by leendertvanwolfswinkel** [comment]
@elisallenens yes I already made that ticket and referred to it in a comment 👍  https://github.com/nens/threedi-api-qgis-client/issues/429

**Comment by elisallenens** [comment]
Discussed this with Martijn Siemerink, and since this ticket would require the modelchecker to optionally call the gridbuilder and generate a grid, it's a whole new feature, not just a simple check. That makes it a little too big for me to pick up right now.
